From 376b31172df353a72bc73f536faa2f5273363f70 Mon Sep 17 00:00:00 2001
From: Lu Yahan <luyahan@qq.com>
Date: Tue, 3 Dec 2024 09:58:29 -0800
Subject: [PATCH 2/2] PR #1788: Fix warning for sign-conversion on riscv

Imported from GitHub PR https://github.com/abseil/abseil-cpp/pull/1788

```
error: operand of ? changes signedness: 'long' to 'uintptr_t' (aka 'unsigned long') [-Werror,-Wsign-conversion]
```

Merge 82deb4bbc42551459d074da2d52cdee5093b8747 into 95950b32c59bc5e4f37883d5610aed8202281336

Merging this change closes #1788

COPYBARA_INTEGRATE_REVIEW=https://github.com/abseil/abseil-cpp/pull/1788 from luyahan:luyahan-patch-1 82deb4bbc42551459d074da2d52cdee5093b8747
PiperOrigin-RevId: 702375910
Change-Id: Ia95d342ddd277dcbde19008e31aaef09f3c3a282
---
 absl/debugging/internal/stacktrace_riscv-inl.inc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/absl/debugging/internal/stacktrace_riscv-inl.inc b/absl/debugging/internal/stacktrace_riscv-inl.inc
index 528ef707..3f9e1240 100644
--- a/absl/debugging/internal/stacktrace_riscv-inl.inc
+++ b/absl/debugging/internal/stacktrace_riscv-inl.inc
@@ -44,7 +44,8 @@ template <typename T>
 static inline ptrdiff_t ComputeStackFrameSize(const T *low, const T *high) {
   const char *low_char_ptr = reinterpret_cast<const char *>(low);
   const char *high_char_ptr = reinterpret_cast<const char *>(high);
-  return low < high ? high_char_ptr - low_char_ptr : kUnknownFrameSize;
+  return low < high ? static_cast<ptrdiff_t>(high_char_ptr - low_char_ptr)
+                    : kUnknownFrameSize;
 }
 
 // Given a pointer to a stack frame, locate and return the calling stackframe,
-- 
2.34.1

